[IMP] sale_stock_prebook: configurable submit booking

vnikolayev1 opened the pull request:
- Added field for sale_stock_prebok route, where user can pick if they want stocks to prebook automatically for this route. In case where it is turned on - stock does not prebooks goods.

vnikolayev1 committed:
[IMP] sale_stock_prebook: configurable submit booking

- Added field for sale_stock_prebok route, where user can pick if they want stocks to prebook automatically for this route. In case where it is turned on - stock does not prebooks goods.

mt-software-de commented on sale_stock_prebook/models/route.py:
```diff
@@ -0,0 +1,11 @@
+        string="No Prebook Stock",
+        help="Prevents stock from prebooking when set."
+             "\nIf not set stock prebooks itself automatically.",
+    )
```
Please rename this to
no_sale_stock_prebook

mt-software-de commented on sale_stock_prebook/models/sale.py:
```diff
@@ -12,6 +12,12 @@ def _prepare_reserve_procurement_values(self, group_id=None):
         return values
 
     def _prepare_reserve_procurement(self, group):
+        no_prebook_routes = self.env["stock.location.route"].search(
+            [("no_prebook_stock", "=", True)]
+        )
+        for route in no_prebook_routes:
+            if route in self.product_id.route_ids:
+                return
```
Please create some own methods. as example
```
# sale.order.line
def _should_prebook_stock(self):
    self.ensure_one()
    for route in self.product_id.route_ids:
        if route.no_sale_stock_prebook:
            return False
    return True

def _prepare_reserve_procurements(self, group):
    procurements = []
    for line in self:
        if not line._should_prebook_stock():
            continue
        procurements.append(line._prepare_reserve_procurement(group))
    return procurements
```

also change then the code in the sale.order to use
_prepare_reserve_procuements

vnikolayev1 committed:
[IMP] sale_stock_prebook: configurable submit booking

- Added field for sale_stock_prebok route, where user can pick if they want stocks to prebook automatically for this route. In case where it is turned on - stock does not prebooks goods.
